checks.d/openstack.py

```diff
@@ -501,10 +616,33 @@ def check(self, instance):
             self.log.debug("Auth Token: %s", self._auth_token)
 
             try:
-                hyp_stats = self.get_hypervisor_stats()
-                self.nova_stats_to_metrics(hyp_stats)
+                if self._local_only:
+                    # The new default: restrict monitoring to this (host, hypervisor, project)
+                    # and it's guest servers
+
+                    hyp = self.get_local_hypervisor()
+                    project = self.get_local_project()
+
+                    aggregate = self.get_aggregates_for_local_host()
+
+                    # Restrict monitoring to hyp and servers, don't do anything else
+                    servers = self.get_servers_managed_by_hypervisor()
+                    host_tags = self._get_tags_for_host()
+
+                    for sid in servers:
+                        server_tags = host_tags + ['host:%s' % sid]
+                        self.get_stats_for_single_server(sid, tags=server_tags)
```

Comment:
We actually catch `409` errors on [line 528](https://github.com/DataDog/dd-agent/blob/dd0917c1223855ddbbb5a61e9abb3405f2de34fc/checks.d/openstack.py#L527). Nova 2.1 API returns a 409 when the instance is powered off, and potentially other error codes for `paused` state. A bare `except Exception` wouldn't hurt though, as we generally want this portion of the check to be completely resilient to failures.